perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -340,21 +355,59 @@ def Run(benchmark_spec):
 
   fio_command = 'sudo %s --output-format=json %s' % (fio.FIO_PATH,
                                                      REMOTE_JOB_FILE_PATH)
+
+  disk = vm.scratch_disks[0]
+
   # TODO(user): This only gives results at the end of a job run
   #      so the program pauses here with no feedback to the user.
   #      This is a pretty lousy experience.
   logging.info('FIO Results:')
-  stdout, stderr = vm.RemoteCommand(fio_command, should_log=True)
 
-  disk = vm.scratch_disks[0]
-  return fio.ParseResults(
-      GetOrGenerateJobFileString(FLAGS.fio_jobfile,
-                                 disk,
-                                 FLAGS.against_device,
-                                 FLAGS.generate_scenarios,
-                                 GetIODepths(FLAGS.io_depths),
-                                 FLAGS.working_set_size),
-      json.loads(stdout))
+  if not FLAGS.generate_scenarios:
+    stdout, stderr = vm.RemoteCommand(fio_command, should_log=True)
+
+    return fio.ParseResults(
+        GetOrGenerateJobFileString(FLAGS.fio_jobfile,
+                                   disk,
+                                   FLAGS.against_device,
+                                   FLAGS.generate_scenarios,
+                                   GetIODepths(FLAGS.io_depths),
+                                   FLAGS.working_set_size),
+        json.loads(stdout))
+  else:
+    # We want to run each scenario for FLAGS.run_for_minutes time.
+    run_reps = FLAGS.run_for_minutes // MINUTES_PER_JOB
+    if FLAGS.run_for_minutes % MINUTES_PER_JOB != 0:
+      run_reps += 1
+      # We already warned the user about rounding during the prepare
+      # phase. No need to warn them again here.
+
+    samples = []
+    start_time = datetime.datetime.now()
+    for rep_num in xrange(run_reps):
+      logging.info('**** Repetition number %s of %s ****', rep_num, run_reps)
+      run_start = datetime.datetime.now()
+      stdout, stderr = vm.RemoteCommand(fio_command, should_log=True)
+
+      seconds_since_start = int(round((run_start - start_time).total_seconds()))
+      minutes_since_start = seconds_since_start // SECONDS_PER_MINUTE
```

Comment:
Done. I think the `round(...)` version is cleaner, but see what you think. I had to add `float(...)` calls to make it round properly.